pandas/io/pytables.py

```diff
@@ -3872,6 +3865,13 @@ def get_blk_items(mgr, blocks):
 
             typ = klass._get_atom(data_converted)
             kind = _dtype_to_kind(data_converted.dtype.name)
+            tz = _get_tz(data_converted.tz) if hasattr(data_converted, "tz") else None
+
+            meta = metadata = ordered = None
+            if is_categorical_dtype(data_converted):
```

Comment:
wouldn't it be better to use a get_atom_categorical like you are doing for other dtypes

Reply:
The get_atom_* funcs are for getting the atom objects; these lines are for getting the meta+metadata+ordered attributes that are going to be pinned to the IndexCol.  Doing it this way just means we're doing it consistently.

That said, ideally I'd like to attach the dtype _object_ rather than attaching `dtype` as a string along with all these other attributes.  That might be an option down the road.